src/topology/algebra/uniform_ring.lean

```diff
@@ -150,8 +156,52 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
-end uniform_space.completion
+@[simp] lemma map_mul_eq_mul_coe (r : R) :
+  completion.map ((*) r) = (*) (r : completion R) :=
+begin
+  ext x,
+  refine completion.induction_on x _ (λ s, _),
+  { exact is_closed_eq (completion.continuous_map) (continuous_mul_left (r : completion R)) },
+  { rw [map_coe (uniform_continuous_const_mul R r), coe_mul] },
+end
 
+instance : algebra R (completion R) :=
+{ commutes' := λ r x, by simp only [ring_hom.to_fun_eq_coe, mul_comm],
+  smul_def' := λ r x, congr_fun (map_mul_eq_mul_coe R r) x,
+  .. (uniform_space.completion.coe_ring_hom : R →+* completion R) }
```

Comment:
I think with #15974 in place you no longer need this instance, as it's just a special case of the one below.